lib/ui/window.dart

```diff
@@ -1338,19 +819,35 @@ enum Brightness {
   light,
 }
 
-/// The [Window] singleton.
+/// The [SingletonFlutterWindow] representing the main window for applications
+/// where there is only one window, such as applications designed for
+/// single-display mobile devices.
 ///
-/// Please try to avoid statically referencing this and instead use a
-/// binding for dependency resolution such as `WidgetsBinding.instance.window`.
+/// For applications that use more than one window, this window typically
+/// represents the first window created, but it may not be visible. Applications
+/// that use more than one window will probably prefer to interact with the
+/// `WidgetsBinding.instance.platformDispatcher` instead.
 ///
-/// Static access of this "window" object means that Flutter has few, if any
+/// Consider avoiding static references to this singleton through
+/// [PlatformDispatcher.instance] and instead prefer using a binding for
+/// dependency resolution such as `WidgetsBinding.instance.window`.
+///
+/// Static access of this `window` object means that Flutter has few, if any
 /// options to fake or mock the given object in tests. Even in cases where Dart
 /// offers special language constructs to forcefully shadow such properties,
 /// those mechanisms would only be reasonable for tests and they would not be
 /// reasonable for a future of Flutter where we legitimately want to select an
 /// appropriate implementation at runtime.
 ///
 /// The only place that `WidgetsBinding.instance.window` is inappropriate is if
-/// a `Window` is required before invoking `runApp()`. In that case, it is
-/// acceptable (though unfortunate) to use this object statically.
-final Window window = Window._();
+/// access to these APIs is required before the binding is initialized by
+/// invoking `runApp()` or `WidgetsFlutterBinding.instance.ensureInitialized()`.
+/// In that case, it is necessary (though unfortunate) to use the
+/// [PlatformDispatcher.instance] object statically.
+///
+/// See also:
+///
+/// * [PlatformDispatcher.views], contains the current list of Flutter windows
+///   belonging to the application, including top level application windows like
+///   this one.
+final Window window = Window._(0, PlatformDispatcher.instance);
```

Comment:
Once the `Window` class is removed, what will be the type of this static?

If it is `SingletonFlutterWindow` the docs above about this being the first window created in a multi-window application wouldn't work, I think, since that window would not be of type `SingletonFlutterWindow`, presumably, since there's by definition more then one window?

If is is of type `FlutterWindow`, than all single-window apps would have to cast this to `SingletonFlutterWindow`, which would be a breaking change for them - and my understanding is that the goal of keeping `SingletonFlutterWindow` around is to avoid that break?

Same question about the future type of `WidgetsBinding.instance.window`.